convert2rhel/checks.py

```diff
@@ -293,7 +295,17 @@ def get_unsupported_kmods(host_kmods, rhel_supported_kmods):
     Ignore certain kmods mentioned in the system configs. These kernel modules moved to kernel core, meaning that the
     functionality is retained and we would be incorrectly saying that the modules are not supported in RHEL."""
     return (
-        host_kmods
-        - rhel_supported_kmods
-        - set(system_info.kmods_to_ignore)
+            host_kmods
+            - rhel_supported_kmods
+            - set(system_info.kmods_to_ignore)
     )
+
+
+def check_repos_are_valid():
+    """Check if all used repositories are valid."""
+    output, ret_code = call_yum_cmd(command="makecache",
+                                    args="--setopt=reposdir=%s -v --setopt=*.skip_if_unavailable=False" % YUM_REPO_DIR)
+    errors_strs = ['error', 'err', 'failure']
+    if ret_code != 0 or any(err in output.lower() for err in errors_strs):
```

Comment:
@fellipeh, have you discussed with @j-mracek the meaning of return codes of dnf makecache? I.e. is it safe to assume that return code 0 means that the repo is accessible and any other return code means there's a problem?

Also, is there a particular reason for looking for strings ['error', 'err', 'failure'] in the makecache command output when it returns 0?

Reply:
I just cheking the list of errors string to be more care, if we have any error but escape 0 as Eric show us above.

Reply:
That was a different command that was printing out errors even when returning 0. The output of `makecache` may be different.

In my testing, when making the baseurl incorrect, I'm getting this on CentOS 7:
```
[root@c2r-20210408201227 ~]# yum makecache -v --setopt=*.skip_if_unavailable=False --disablerepo="*" --enablerepo=base
Loading "fastestmirror" plugin
Config time: 0.006
Yum version: 3.4.3
Loading mirror speeds from cached hostfile
Could not retrieve mirrorlist http://badurl/ error was
14: curl#6 - "Could not resolve host: badurl; Unknown error"
 * base: mirror.karneval.cz
base                                                                                                                                                                                                                   | 3.6 kB  00:00:00     
(1/2): base/other_db                                                                                                                                                                                                   | 2.6 MB  00:00:02     
(2/2): base/filelists_db                                                                                                                                                                                               | 7.2 MB  00:00:03     
Metadata Cache Created
[root@c2r-20210408201227 ~]# echo $?
0
```
The reason for that is that it uses cached metadata. So, rather than relying on a certain keywords we just assume might appear, it seems to me better to clean the yum cache first. Then every inaccessible url or non-existing repoid ends up with non-0 return code.
```
[root@c2r-20210408201227 ~]# yum clean metadata
Loaded plugins: fastestmirror
Cleaning repos: base copr:copr.fedorainfracloud.org:group_oamg:convert2rhel extras updates
12 metadata files removed
12 sqlite files removed
0 metadata files removed
[root@c2r-20210408201227 ~]# yum makecache -v --setopt=*.skip_if_unavailable=False --disablerepo="*" --enablerepo=base
Loading "fastestmirror" plugin
Config time: 0.005
Yum version: 3.4.3
Loading mirror speeds from cached hostfile
Could not retrieve mirrorlist http://badurl/ error was
14: curl#6 - "Could not resolve host: badurl; Unknown error"


 One of the configured repositories failed (Unknown),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Run the command with the repository temporarily disabled
            yum --disablerepo=<repoid> ...

     4. Disable the repository permanently, so yum won't use it by default. Yum
        will then just ignore the repository until you permanently enable it
        again or use --enablerepo for temporary usage:

            yum-config-manager --disable <repoid>
        or
            subscription-manager repos --disable=<repoid>

     5. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=<repoid>.skip_if_unavailable=true

Cannot find a valid baseurl for repo: base
```

Reply:
I already remove the str checks and check the return code.